packages/hathor-rpc-handler/src/rpcMethods/getBalance.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet } from '@hathor/wallet-lib';
+import { GetBalanceObject } from '@hathor/wallet-lib/lib/wallet/types';
+import {
+  ConfirmationPromptTypes,
+  GetBalanceConfirmationPrompt,
+  GetBalanceRpcRequest,
+  PromptHandler,
+} from '../types';
+import { NotImplementedError, PromptRejectedError } from '../errors';
+import { validateNetwork } from '../helpers';
+
+/**
+ * Gets the balance for specified tokens using the provided wallet.
+ *
+ * @param rpcRequest - The RPC request containing the parameters for getting the balance.
+ * @param wallet - The wallet instance to use for retrieving the balance.
+ * @param promptHandler - A function to handle prompts for user confirmation.
+ *
+ * @returns The balances of the specified tokens.
+ *
+ * @throws {NotImplementedError} - If address indexes are specified, which is not implemented.
+ * @throws {PromptRejectedError} - If the user rejects the balance confirmation prompt.
+ */
+export async function getBalance(
+  rpcRequest: GetBalanceRpcRequest,
+  wallet: HathorWallet,
+  promptHandler: PromptHandler,
+) {
+  const { network, tokens, addressIndexes } = rpcRequest.params;
+
+  if (addressIndexes) {
```

Comment:
Why? We have a method in the wallet `getAddressInfo` that has the balance.

Reply:
`getAddressInfo` is only available in the old facade, we can implement it in the new facade, but I suggest we leave this not implemented in this PR

We can do it in part 3/5